src/js/app.js

```diff
@@ -0,0 +1,113 @@
+
+document.addEventListener("click", activeTodoItem);
+document.addEventListener("click", allTodoItem);
+document.addEventListener("click", finishTodoItem);
+
+function renderTodoItemTemplate(title) {
+    return ` <li class="active">
+                  <div class="view">
+                      <input class="toggle" onclick="onToggleTodoItem(event)" type="checkbox" >
+                      <label class="label" ondblclick="updateTodoItem(event)">${title}</label>
+                      <button class="destroy" onclick="deleteTodoItem(event)"></button>
+                  </div>
+                  <input class="edit" value=${title} />
+              </li>`;
+}
+
+function addTodoItem(event) {
+    const todoTitle = event.target.value;
+    const todoList = document.getElementById("todo-list");
+    if ("Enter" === event.key && "" !== todoTitle) {
+        todoList.insertAdjacentHTML("beforeend", renderTodoItemTemplate(todoTitle));
+        event.target.value = "";
+        count();
+    }
+}
+
+function onToggleTodoItem(event) {
+    const target = event.target;
+    if ("toggle" === target.className) {
+        if ("completed" === target.closest("li").className) {
+            target.closest("li").className = "active";
+        } else {
+            target.closest("li").className = "completed";
+        }
+    }
+}
+
+function updateTodoItem(event) {
+    const target = event.target;
+    if ("label" === target.className) {
+        target.closest("li").classList.add("editing");
+
+        const editInput = document.querySelector(".edit");
+        editInput.addEventListener("keydown", function(event){
+            if ("Escape" === event.key) {
+                target.closest("li").classList.remove("editing");
+            }
+
+            if ("Enter" === event.key) {
+                target.innerText = editInput.value;
+                target.closest("li").classList.remove("editing");
+            }
+        })
+    }
+}
+
+function deleteTodoItem(event) {
+    const target = event.target;
+    if ("destroy" === target.className) {
+        target.closest("li").remove();
+    }
+}
+
+function activeTodoItem(event) {
+    if (event.target.className === 'active') {
+        let countItems = 0;
+        const todoList = $toggleInput.children;
+        for (let i = 0; i < todoList.length; i++) {
+            if (todoList[i].classList.contains("completed")) {
+                todoList[i].style.display = "none";
+            } else {
+                countItems++;
+                todoList[i].style.display = "block";
+            }
+        }
+        $todoCount.innerHTML = countItems.toString();
+    }
+}
+
+function allTodoItem(event) {
+    if (event.target.className === 'all selected') {
+        const childrens = $toggleInput.children;
+        for (let i = 0; i < childrens.length; i++) {
+            childrens[i].style.display = "block";
+        }
+        $todoCount.innerHTML = $toggleInput.children.length.toString();
```

Comment:
JS에서는 자동형변환이 일어나므로, 반드시 String으로 변환시키지 않으셔도 됩니다! 
(사실 저는 이걸 보기전에 toString 을 사용하여 String으로 변환할 생각 자체를 못해봤네요;; 너무나 동적 타입에 익숙해져버려서...)

```suggestion
        $todoCount.innerHTML = $toggleInput.children.length;
```